src/jgikbase/test/idmapping/userhandlers/kbase_user_handler_test.py

```diff
@@ -56,17 +57,30 @@ def test_get_user_fail_not_json():
                       IOError('Non-JSON response from KBase auth server, status code: 404'))
 
 
-def test_get_user_fail_auth_returned_error():
+def test_get_user_fail_invalid_token():
     with requests_mock.Mocker() as m:
         m.get('http://my1stauthservice.com/api/api/V2/token',
               request_headers={'Authorization': 'bar'},
               status_code=401,
-              json={'error': {'message': '10020 Invalid token'}})
+              json={'error': {'appcode': 10020, 'message': '10020 Invalid token'}})
+
+        kbuh = KBaseUserHandler('http://my1stauthservice.com/api', Token('foo'))
+
+        fail_get_user(kbuh, Token('bar'), InvalidTokenError(
+            'KBase auth server reported token is invalid.'))
+
+
+def test_get_user_fail_auth_returned_other_error():
+    with requests_mock.Mocker() as m:
```

Comment:
Down the line you might one 1 or 2 small integration tests that actually run the full auth

Reply:
Hmmm, that'd require getting the auth server to run locally, which is doable (the search service does this), but it'd be a lot of work for python. I'd *like* to do it, but out of all the stuff that needs to be done that's pretty low on the list.

Reply:
Can you just have an integration case that makes requests to an auth URL set by an env var, and when you run tests, you can choose to set it to a local server, CI, or wherever?

Reply:
No, because you'd need valid auth tokens to run tests, which means you'd need to put them in travis secure variables, which means tests no longer work on PRs.